buildenv/jenkins/JenkinsfileBase

```diff
@@ -805,19 +805,25 @@ def run_parallel_tests() {
 			} else {
 				waitForANodeToBecomeActive("$LABEL", "0")
 			}
-			node("$LABEL") {	
+			node("$LABEL") {
+				def buildResult = ""	
 				childJobs.each {
 					cjob ->
 					def jobInvocation = cjob.value.getRawBuild()
 					def buildId = jobInvocation.getNumber()
 					def name = cjob.value.getProjectName()
+					def childResult = cjob.value.getCurrentResult()
 					try {
+						echo "${name} #${buildId} completed with status ${childResult}"
 						copyArtifacts (projectName: "${name}", selector: specific("${buildId}"), filter: "**/TestTargetResult.tap", target:"${name}/${buildId}")
+						step([$class: "TapPublisher", testResults: "${name}/${buildId}/**/TestTargetResult.tap", outputTapToConsole: false, failIfNoResults: false])
 					} catch (Exception e) {
 						echo "Cannot copy TestTargetResult.tap from ${name} with buildid ${buildId} . Skipping copyArtifacts..."
+						echo "set build status to ${childResult}"
+						buildResult = childResult
 					}
 				}
-				step([$class: "TapPublisher", testResults: "**/TestTargetResult.tap", outputTapToConsole: false, failIfNoResults: false])
+				currentBuild.result = buildResult
```

Comment:
We do not reset `currentBuild.result` in normal cases. We should only set `currentBuild.result` if buildResult is not empty. Also, line 822 should be moved here.

Reply:
updated :)